pkg/secretless/config/v2/config.go

```diff
@@ -47,3 +51,115 @@ func NewConfig(v2YAML []byte) (*Config, error) {
 		Services: services,
 	}, nil
 }
+
+// NewConfigsByType converts a slice of v2.Service configs into the configs
+// needed to actually created ProxyServices -- configsByType.  In particular, it
+// takes all the http configs and creates proper HTTPServiceConfig objects out
+// of them -- grouping the raw v2.Service configs by their listenOn property.
+// The remaining services are tcp, and already correspond 1-1 to the services
+// we'll run.
+// TODO: Eventually the application code should not be dealing directly with
+//   []Service at all, but the processing into these more appropriate domain
+//   configs should occur entirely at the border.
+func NewConfigsByType(
+	uncheckedConfigs []*Service,
+	availPlugins plugin.AvailablePlugins,
+) ConfigsByType {
+
+	// Get the nil checks out of the way.
+	var rawConfigs []Service
+	for _, cfg := range uncheckedConfigs {
+		if cfg == nil {
+			// Hard-coding log here is okay since nils should never occur and
+			// we won't be unit testing this.  Instead, we'll make the change
+			// in the TODO_ above and this code will be deleted then.
+			log.Fatalln("Nil configuration is not allowed!")
```

Comment:
This is a convenience function called quite deep in the stack. I don't think it should have this catastrophic side effect. I think that should bubble up to a place where it's more obvious what the issue.

Reply:
Ah, so remember per some of my other TODOs, all this stuff will eventually happen at the border.  So it would be fail fast code, exactly where want it, once we make that change.  I didn't want to make that change yet, however, as it's... not hard exactly, but a can of worms in the same way you updating all the application code to use v2 instead of v1 was.  

TLDR: I agree with you assessment, and plan to solve it, but I think solving it later is the right play.